pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -61,13 +63,41 @@ makeWrapper() {
             command="${params[$((n + 1))]}"
             n=$((n + 1))
             echo "$command" >> "$wrapper"
-        elif [[ ("$p" == "--suffix") || ("$p" == "--prefix") ]]; then
+        elif [[ ("$p" == "--suffix") || ("$p" == "--prefix") || ("$p" == "--suffix-dedup") || ("$p" == "--prefix-dedup") ]]; then
             varName="${params[$((n + 1))]}"
             separator="${params[$((n + 2))]}"
             value="${params[$((n + 3))]}"
             n=$((n + 3))
             if test -n "$value"; then
-                if test "$p" = "--suffix"; then
+                if [[ "$p" == *'-dedup' ]]; then
+                    {
+                        # This abomination removes all occurences of the value that is to be prepended/appended.
+                        # We add (or re-add) the value later. This ensures it's always the last or first value.
+                        # As an example, adding /bin to a colon-separated environment variable would remove:
+                        # - /bin:*
+                        # - *:/bin:*
+                        # - *:/bin
+                        echo "OLDIFS=\$IFS"
+                        echo "value=${value@Q}"
+                        echo "IFS=${separator@Q}"
+                        echo "for v in \$value; do"
+                        echo "  tmp="
+                        echo "  for e in \$$varName; do"
```

Comment:
You can do something like `[[ \$$varName == *$separator$v$separator* ]]` to figure out if the value is already present. You can remove the loop and be a lot faster.

Reply:
Globs aren't flexible enough to express all possible variations that should be possible. See the discussion in the previous PR. Also, I don't think there's reason for concern over performance here - the current loop handles an artificially long list of items with three values to deduplicate in ~1ms on my machine.

Reply:
I had this code in mind

https://github.com/NixOS/nixpkgs/blob/e2508a8e57eb8a6478f2b39fc5c7106654cfea41/pkgs/stdenv/generic/setup.sh#L160-L163

which uses normal globs. (Incidentally, I think `--suffix` should behave like `addToSearchPath`.)

Reply:
That doesn't do the same thing, though. If you _don't_ want deduplication, globs are probably fine.